Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,438 @@
+				{
+					Name = summary,
+					Start = startTime.UtcTime,
+					End = endTime.UtcTime,
+					Location = location,
+					Description = description,
+					AllDay = Convert.ToBoolean(IsAllday),
+					ExternalID = Id.ToString(),
+				};
+			}
+			finally
+			{
+				calRecord?.Dispose();
+				calRecord = null;
+
+				calManager?.Dispose();
+				calManager = null;
+			}
+			return Task.FromResult(calEvent);
+		}
+
+		/// <summary>
+		/// Creates a new calendar or updates the name and color of an existing one.
+		/// </summary>
+		/// <param name="calendar">The calendar to create/update</param>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device or is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		/// <exception cref="System.PlatformNotSupportedException">Platform does not support api</exception>
+		public Task AddOrUpdateCalendarAsync(Calendar calendar)
+		{
+			throw new PlatformNotSupportedException("Tizen platform does not support add or update calendars");
+		}
+
+		/// <summary>
+		/// Add new event to a calendar or update an existing event.
+		/// Throws if Calendar ID is empty, calendar does not exist, or calendar is read-only.
+		/// </summary>
+		/// <param name="calendar">Destination calendar</param>
+		/// <param name="calendarEvent">Event to add or update</param>
+		/// <exception cref="System.ArgumentException">Calendar is not specified, does not exist on device, or is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="System.InvalidOperationException">Editing recurring events is not supported</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public async Task AddOrUpdateEventAsync(Calendar calendar, CalendarEvent calendarEvent)
+		{
+			Calendar existingCal = null;
+
+			if (!string.IsNullOrEmpty(calendar.ExternalID) || !string.IsNullOrEmpty(calendarEvent.ExternalID))
+			{
+				existingCal = await GetCalendarByIdAsync(calendar.ExternalID).ConfigureAwait(false);
+			}
+
+			if (calendarEvent.End < calendarEvent.Start)
+			{
+				throw new ArgumentException("End time may not precede start time", "calendarEvent");
+			}
+
+			if (!existingCal.CanEditEvents)
+			{
+				throw new ArgumentException("Cannot delete event from readonly calendar", "calendar");
+			}
+			else
+			{
+				CalendarManager calManager = null;
+				CalendarRecord updateRecord = null;
+
+				try
+				{
+					calManager = new CalendarManager();
+					updateRecord = calManager.Database.Get(Event.Uri, Convert.ToInt32(calendarEvent.ExternalID));
+
+					if (updateRecord != null)
+					{
+						try
+						{
+							DateTime startTime = calendarEvent.AllDay
```

Comment:
This checks the `AllDay` property of `CalendarEvent` when setting the `start`/`end` times, but it never actually sets the `AllDay` flag of the Tizen `CalendarRecord`

(nor does the `addRecord` version, although that one doesn't have this check for the start/end times... which itself seems odd and inconsistent...)